typescript/web/src/components/export-button/modal/index.tsx

```diff
@@ -78,64 +68,27 @@ export const exportCocoDataset = async ({
     .join("-")}T${String(dateObject.getHours()).padStart(2, "0")}${String(
     dateObject.getMinutes()
   ).padStart(2, "0")}${String(dateObject.getSeconds()).padStart(2, "0")}`;
-  const datasetName = `dataset-${date}-Coco`;
-  if (options.exportImages) {
-    const {
-      data: { images },
-    } = await client.query({ query: getImagesQuery, variables: { datasetId } });
-    const zip = new JSZip();
-    zip.file(
-      `${datasetName}/annotations.json`,
-      exportToCoco.substr(exportToCoco.indexOf(",") + 1),
-      {
-        base64: true,
-      }
-    );
-    await Promise.all(
-      images.map(
-        async ({
-          id,
-          name,
-          url,
-          mimetype,
-        }: {
-          id: string;
-          name: string;
-          url: string;
-          mimetype: string;
-        }) => {
-          const dataUrl = await (async (): Promise<string> => {
-            const blob = await fetch(url).then((r) => r.blob());
-            return new Promise((resolve) => {
-              const reader = new FileReader();
-              reader.onload = () => resolve(reader.result as string);
-              reader.readAsDataURL(blob);
-            });
-          })();
-          zip.file(
-            `${datasetName}/images/${name}_${id}.${mime.extension(mimetype)}`,
-            dataUrl.substr(dataUrl.indexOf(",") + 1),
-            {
-              base64: true,
-            }
-          );
-        }
-      )
-    );
-    const blobZip = await zip.generateAsync({ type: "blob" });
-    const url = window.URL.createObjectURL(blobZip);
-    const element = document.createElement("a");
-    element.href = url;
-    element.download = `${datasetName}.zip`;
-    setIsExportRunning(false);
-    element.click();
-  } else {
-    const element = document.createElement("a");
-    element.href = exportToCoco;
-    element.download = `${datasetName}.json`;
-    setIsExportRunning(false);
-    element.click();
-  }
+  const datasetName = `dataset-${date}-${format.toLowerCase()}`;
+  const {
+    data: { exportDataset: exportDatasetUrl },
+  } = await client.query({
+    query: exportQuery,
+    variables: {
+      datasetId,
+      format,
+      options: {
+        coco: { ...options.coco, name: datasetName },
+        yolo: { ...options.yolo, name: datasetName },
+      },
```

Comment:
Nitpicking, feel free to ignore:

We could pass only the options of the chosen format, it would make the request a bit lighter.